cvmfs/dns.cc

```diff
@@ -68,10 +68,22 @@ static void PinpointHostSubstr(
     }
   }
 
+  // Search '@' within the hostname part and jump behind if present
+  if (*pos_begin > 0) {
+    for (i = *pos_begin; i < len; ++i) {
+      if (url[i] == '/') {
+	break;
+      } else if (url[i] == '@') {
+	*pos_begin = ++i;
```

Comment:
Same here

Reply:
Do we need to check here that `pos_begin` does not go beyond the string? I think we should have a test with `http://user:pw@` as an input.

Reply:
The character at `url[i]` is guaranteed here to be `'@'` (and therefore not the string terminator), so the subsequent character at `url[++i]` is guaranteed not to be beyond the end of the string.  But I'll add a test case for `http://user:pw@` anyway